src/BTree.mo

```diff
@@ -0,0 +1,139 @@
+/// Imperative sequences as B-Trees.
+
+import A "Array";
+import I "Iter";
+import List "List";
+import Option "Option";
+import Order "Order";
+import P "Prelude";
+import Debug "Debug";
+import Prim "mo:⛔";
+
+module {
+
+  /// Constants we use to shape the tree.
+  /// See https://en.wikipedia.org/wiki/B-tree#Definition
+  module Constants {
+    let MAX_CHILDREN = 4;
+  };
+
+  public type Compare<K> = {
+    show : K -> Text;
+    compare : (K, K) -> Order.Order
+  };
+
+  public type Data<K, V> = [(K, V)];
+
+  public type Internal<K, V> = {
+    data : Data<K, V>;
+    trees : [Tree<K, V>];
+  };
+
+  public type Tree<K, V> = {
+    #internal : Internal<K, V>;
+    #leaf : Data<K, V>;
+  };
+
+  func find_data<K, V>(data : Data<K, V>, find_k : K, c : (K, K) -> Order.Order) : ?V {
+    for ((k, v) in data.vals()) {
+      if (c(k, find_k) == #equal) { return ?v };
+    };
+    return null
+  };
+
+  public func find<K, V>(t : Tree<K, V>, k : K, c : (K, K) -> Order.Order) : ?V {
+   switch t {
+      case (#leaf(d)) { return find_data<K, V>(d, k, c) };
+      case (#internal(i)) {
+        for (j in I.range(0, i.data.size())) {
+          switch (c(k, i.data[j].0)) {
+            case (#equal) { return ?i.data[j].1 };
+            case (#less) { return find<K, V>(i.trees[j], k, c) };
+            case _ { }
+          }
+        };
+        find<K, V>(i.trees[i.data.size()], k, c)
+      };
+    };
+  };
+
+  /// Check that a B-Tree instance observes invariants of B-Trees.
```

Comment:
If this module is just for testing/debugging, should `Check` be in a different file?

When I import the default BTree module like `import BTree "mo:base/BTree";`, does it shake out the `Check`, or does it include it and bloat the import size?

Reply:
> If this module is just for testing/debugging, should Check be in a different file?

Why?  It's more enclosed and encapsulated here.  There is no way to "hide" a top-level module in `base`.